code/game/machinery/limbgrower.dm

```diff
@@ -0,0 +1,232 @@
+			dat = search_win(user)
+*/
+	var/datum/browser/popup = new(user, "Limb Grower", name, 400, 500)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/limbgrower/deconstruction()
+	for(var/obj/item/weapon/reagent_containers/glass/G in component_parts)
+		reagents.trans_to(G, G.reagents.maximum_volume)
+
+/obj/machinery/limbgrower/attackby(obj/item/O, mob/user, params)
+	if (busy)
+		user << "<span class=\"alert\">The Limb Grower is busy. Please wait for completion of previous operation.</span>"
+		return 1
+/*
+	if (ispath(O,/obj/item/weapon/reagent_containers))
+		if(!O.reagents.hasreagent("synthflesh") && O.reagents.reagentlist.length!=1)
+			user << "<span class=\"alert\">The Limb grower refuses the chemicals, perhaps it only accepts pure synthflesh?</span>"
+*/
+	if(default_deconstruction_screwdriver(user, "limbgrower_panelopen", "limbgrower_idleoff", O))
+		updateUsrDialog()
+		return
+
+	if(exchange_parts(user, O))
+		return
+
+	if(panel_open)
+		if(istype(O, /obj/item/weapon/crowbar))
+			default_deconstruction_crowbar(O)
+			return 1
+
+	if(user.a_intent == "harm") //so we can hit the machine
+		return ..()
+
+	if(stat)
+		return 1
+
+	if(O.flags & HOLOGRAM)
+		return 1
+
+/obj/machinery/limbgrower/Topic(href, href_list)
+	if(..())
+		return
+	if (!busy)
+		if(href_list["menu"])
+			screen = text2num(href_list["menu"])
+
+		if(href_list["category"])
+			selected_category = href_list["category"]
+
+		if(href_list["make"])
+
+			/////////////////
+			//href protection
+			being_built = files.FindDesignByID(href_list["make"]) //check if it's a valid design
+			if(!being_built)
+				return
+
+
+			var/synth_cost = being_built.reagents["synthflesh"]*prod_coeff
+			var/power = max(2000, synth_cost/5)
+
+			if(reagents.has_reagent("synthflesh", being_built.reagents["synthflesh"]*prod_coeff))
+				busy = 1
+				use_power(power)
+				flick("limbgrower_fill",src)
+				icon_state = "limbgrower_idleon"
+				spawn(32*prod_coeff)
+					use_power(power)
+					reagents.remove_reagent("synthflesh",being_built.reagents["synthflesh"]*prod_coeff)
+					var/B = being_built.build_path
+					world << "Starting creaton of a limb"
+					if(ispath(B, /obj/item/bodypart))	//This feels like spatgheti code, but i need to initilise a limb somehow
+						//i need to create a body part manually using a set specias dna
+						var/obj/item/bodypart/L
+						world << "category start"
+						world << "[selected_category] is the current category"
+						switch(selected_category)
+							if("Human")
+								//Human dna
+								L = new B()
+								world << "Trying to update limb"
+								L.name = "Synthentic Human Limb"
+								L.desc = "A synthentic human limb that will morph on its first use in surgery"
+							else if("Lizard")
+								//Lizard dna
+								L = new B()
+								world << "Trying to update lizard limb"
+								L.name = "Synthentic Lizard Limb"
+								L.desc = "A synthentic lizard limb that will morph on its first use in surgery"
```

Comment:
No it won't, you can't use invalid bodyparts in invalid locations
arm in leg slot = failure.